interfaces/inc/c_pal/uuid.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef UUID_H
+#define UUID_H
+
+#ifdef WIN32 
+#include "windows.h"
+#endif
+
+typedef unsigned char UUID_T[16]; /*introduces UUID_T as "array of 16 bytes"*/
+
+#include "macro_utils/macro_utils.h"
+#include "umock_c/umock_c_prod.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    MOCKABLE_FUNCTION(, int, uuid_produce, UUID_T, destination);
+
+#ifdef WIN32 /*some functions only exists in Windows realm*/
+    MOCKABLE_FUNCTION(, int, uuid_from_GUID, UUID_T, destination, const GUID*, source);
+#endif

```

Comment:
The pattern I think we've been using is to have a uuid_win32.h in the win32/inc/c_pal folder with Windows-only things.

Whoever wants to use this would have to include uuid_win32.h which is nice because it makes it more obvious that the caller is (currently) Windows-only, like seeing a "windows.h" include.

/consider

Reply:
I'll think about it.

Reply:
It is nice to make it explicit, but as a user of c_pal, do I really want to include more than 1 header? Why shouldn't I get eeeeeverrrything in 1 line, as opposed to hunting down "which other header provides the functionality which I want". 

At the moment I see this mostly as a style issue. There's no performance hit, there's no other technical reason that would warrant splitting the # include in 2 separate headers. I will keep it as is until it starts to hurt somehow and then we'll reanalyze.

I also asked Jelani in a double-blind style and he said he wants to have just 1 header. So it stays as is.